client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -242,11 +243,14 @@ class StatsPeriodNavigation extends PureComponent {
 
 		const path = `/stats/${ period }/${ slug }`;
 		const url = getPathWithUpdatedQueryString(
-			{ chartStart, chartEnd, navigation: 'arrow' },
+			{ chartStart, chartEnd, navigation: NAVIGATION_METHOD_ARROW },
 			path
 		);
 
-		page( url );
+		// Redirect to the page by replacing the current history entry, 
+		// rather than pushing a new one. This ensures the navigation 
+		// does not create an additional entry in the browser's history stack.
+		page.redirect( url );
```

Comment:
I [encountered a concern](https://github.com/Automattic/wp-calypso/pull/101239#discussion_r1992450697) when ignoring history while navigating with the arrows. I think it is still a trade-off here between considering another approach. cc @kangzj

Reply:
Could you share more about the reason we'd like not to push a new entry please?